frontend/src/app/index.tsx

```diff
@@ -36,31 +47,60 @@ export default class Index extends Component {
   }
 
   private getRows = (event) => {
-    console.log("what do I see? " + event.target.value)
-    this.setState({
-      input: event.target.value
-    }, () => {
-      console.log("Now I get the expected value down " + this.state.input)
-      var backend = "/modules.json?search="
-      if (this.state.input != null && this.state.input != "" && this.state.input != "*") {
-        backend = backend + this.state.input
-        console.log(backend)
-      }
-      fetch(backend)
-        .then(response => response.json())
-        .then(responseJSON => this.setState({ data: responseJSON }))
-        .then(() => console.log("JSON string is " + JSON.stringify(this.state.data)))
-    })
+    if (event.key === 'Enter') {
+      console.log("what do I see? " + event.target.value)
+      this.setState({
+        input: event.target.value
+      }, () => {
+        console.log("Now I get the expected value down " + this.state.input)
+        var backend = "/modules.json?search="
+        if (this.state.input != null && this.state.input != "" && this.state.input != "*") {
+          backend = backend + this.state.input
+          console.log(backend)
+        }
+        fetch(backend)
+          .then(response => response.json())
+          .then(responseJSON => this.setState({ data: responseJSON }))
+          .then(() => {
+           // console.log("JSON string is " + JSON.stringify(this.state.data))
+            if (JSON.stringify(this.state.data) == "[]"){
+              this.setState({isEmptyResults: true,
+              data: [{ "pant:transientPath": '', "jcr:created": '', "name": "search_intro_message_thn", "jcr:title": "Search is case sensitive. Type '*' and press 'Enter' for all modules.", "jcr:description": "", "sling:transientSource": "", "pant:transientSourceName": "" }],
+              })
+            } else {
+              this.setState({isEmptyResults: false})
+            }
+          })
+      })
+    }
   };
 
-
   private setPreview(path: string) {
     console.log("what do I see when you click ? " + path)
-    if (path != "") {      
+    if (path != "") {
       return window.open("/" + path + ".preview");
     } else {
       return ""
     }
   };
 
+  private formatDate(date: Date) {
+    //2019/05/07 14:21:36
+    var dateStr = date.getFullYear().toString() + "/" +
+      date.getMonth().toString() + "/" +
+      date.getDay().toString() + " " +
+      date.getHours().toString() + ":" +
+      date.getMinutes().toString() + ":" +
+      date.getSeconds().toString()
```

Comment:
Curious if there's a react equivalent to java's SimpleDateFormatter, but this certainly works.

Reply:
There are external libraries. That do date formatting, but for this simple format I tried to avoid adding a dependency. If we need something more complex maybe we can look at something like:
https://www.npmjs.com/package/react-moment
https://stackoverflow.com/questions/34590369/formatting-a-date-string-in-react-native